### PR TITLE
Folder cleanup

### PR DESCRIPTION
**Pull request by charmander**

### CODE REVIEW

File: weasyl/folder.py [hunk 1]
```diff
@@ -155,78 +157,51 @@ def select_preview(userid, otherid, rating):
...
-    else:
-        query = d.execute(
-            "SELECT folderid, title, parentid FROM folder WHERE userid = %i AND settings !~ 'h' ORDER BY parentid, title",
-            [userid])
-
-        for i in range(len(query)):
-            if query[i][2]:
-                break
+    for top_level_folder in result:
+        children = folders_by_parentid.get(top_level_folder.folderid)
```
Comment (kfkitsune) [inline weasyl/folder.py:179]: I don't know if this is an artifact of me starting from a completely empty database or not (via manual Alembic migrating the DB from blank), but when creating a folder with no folders already, this line raises the following:
```
2020-08-18T02:24:10+0000 [stdout#info] unhandled error (request id None) in {'HTTP_ACCEPT': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8', 'webob._parsed_query_vars': (GET([]), ''), 'CONTENT_TYPE': '', 'HTTP_REFERER': 'https://lo.weasyl.com:8443/manage/folders', 'HTTP_X_FORWARDED_PROTO': 'https', 'wsgi.multithread': True, 'SCRIPT_NAME': '', 'HTTP_X_REQUEST_STARTED_AT': '1597717450.93344688', 'REQUEST_METHOD': 'GET', 'HTTP_HOST': 'lo.weasyl.com:8443', 'PATH_INFO': '/manage/folders', 'SERVER_PROTOCOL': 'HTTP/1.0', 'QUERY_STRING': '', 'HTTP_UPGRADE_INSECURE_REQUESTS': '1', 'HTTP_X_REAL_IP': '10.0.2.2', 'CONTENT_LENGTH': '', 'HTTP_USER_AGENT': 'Mozilla/5.0 (X11; Linux x86_64; rv:68.0) Gecko/20100101 Firefox/68.0', 'HTTP_CONNECTION': 'close', 'HTTP_COOKIE': 'WZL=osxqZQa7NjliBdEJbooKUZVCLwINRPMOOFja7xoR4x8AlMVNTszDUH3LodHcFaSI; __cfduid=df5cd83296b157b956e682775c1454aa21597716017', 'SERVER_NAME': 'lo.weasyl.com', 'REMOTE_ADDR': '127.0.0.1', 'HTTP_X_FORWARDED_FOR': '10.0.2.2', 'wsgi.errors': <twisted.web.wsgi._ErrorStream object at 0x7f390bbddf10>, 'wsgi.multiprocess': False, 'HTTP_ACCEPT_LANGUAGE': 'en-US,en;q=0.5', 'wsgi.input': <weasyl.middleware.InputWrap object at 0x7f390bb93e50>, 'wsgi.version': (1, 0), 'wsgi.url_scheme': 'https', 'wsgi.run_once': False, 'SERVER_PORT': '8880', 'HTTP_ACCEPT_ENCODING': 'gzip, deflate, br', 'webob._parsed_cookies': ({u'WZL': u'osxqZQa7NjliBdEJbooKUZVCLwINRPMOOFja7xoR4x8AlMVNTszDUH3LodHcFaSI', u'__cfduid': u'df5cd83296b157b956e682775c1454aa21597716017'}, 'WZL=osxqZQa7NjliBdEJbooKUZVCLwINRPMOOFja7xoR4x8AlMVNTszDUH3LodHcFaSI; __cfduid=df5cd83296b157b956e682775c1454aa21597716017')}
2020-08-18T02:24:10+0000 [stderr#error] Traceback (most recent call last):
2020-08-18T02:24:10+0000 [stderr#error]   File "/vagrant/weasyl-env/local/lib/python2.7/site-packages/pyramid/tweens.py", line 41, in excview_tween
2020-08-18T02:24:10+0000 [stderr#error]     response = handler(request)
2020-08-18T02:24:10+0000 [stderr#error]   File "/vagrant/weasyl/middleware.py", line 161, in query_debug_tween
2020-08-18T02:24:10+0000 [stderr#error]     return handler(request)
2020-08-18T02:24:10+0000 [stderr#error]   File "/vagrant/weasyl/middleware.py", line 233, in http2_server_push
2020-08-18T02:24:10+0000 [stderr#error]     resp = handler(request)
2020-08-18T02:24:10+0000 [stderr#error]   File "/vagrant/weasyl/middleware.py", line 193, in database_session_cleanup_tween
2020-08-18T02:24:10+0000 [stderr#error]     return handler(request)
2020-08-18T02:24:10+0000 [stderr#error]   File "/vagrant/weasyl/middleware.py", line 44, in cache_clear_tween
2020-08-18T02:24:10+0000 [stderr#error]     return handler(request)
2020-08-18T02:24:10+0000 [stderr#error]   File "/vagrant/weasyl/middleware.py", line 63, in db_timer_tween
2020-08-18T02:24:10+0000 [stderr#error]     resp = handler(request)
2020-08-18T02:24:10+0000 [stderr#error]   File "/vagrant/weasyl/middleware.py", line 117, in session_tween
2020-08-18T02:24:10+0000 [stderr#error]     return handler(request)
2020-08-18T02:24:10+0000 [stderr#error]   File "/vagrant/weasyl/middleware.py", line 180, in status_check_tween
2020-08-18T02:24:10+0000 [stderr#error]     return handler(request)
2020-08-18T02:24:10+0000 [stderr#error]   File "/vagrant/weasyl-env/local/lib/python2.7/site-packages/pyramid/router.py", line 148, in handle_request
2020-08-18T02:24:10+0000 [stderr#error]     registry, request, context, context_iface, view_name
2020-08-18T02:24:10+0000 [stderr#error]   File "/vagrant/weasyl-env/local/lib/python2.7/site-packages/pyramid/view.py", line 667, in _call_view
2020-08-18T02:24:10+0000 [stderr#error]     response = view_callable(context, request)
2020-08-18T02:24:10+0000 [stderr#error]   File "/vagrant/weasyl-env/local/lib/python2.7/site-packages/pyramid/config/views.py", line 188, in attr_view
2020-08-18T02:24:10+0000 [stderr#error]     return view(context, request)
2020-08-18T02:24:10+0000 [stderr#error]   File "/vagrant/weasyl-env/local/lib/python2.7/site-packages/pyramid/config/views.py", line 214, in predicate_wrapper
2020-08-18T02:24:10+0000 [stderr#error]     return view(context, request)
2020-08-18T02:24:10+0000 [stderr#error]   File "/vagrant/weasyl-env/local/lib/python2.7/site-packages/pyramid/viewderivers.py", line 401, in viewresult_to_response
2020-08-18T02:24:10+0000 [stderr#error]     result = view(context, request)
2020-08-18T02:24:10+0000 [stderr#error]   File "/vagrant/weasyl-env/local/lib/python2.7/site-packages/pyramid/viewderivers.py", line 144, in _requestonly_view
2020-08-18T02:24:10+0000 [stderr#error]     response = view(request)
2020-08-18T02:24:10+0000 [stderr#error]   File "/vagrant/weasyl/controllers/decorators.py", line 21, in inner
2020-08-18T02:24:10+0000 [stderr#error]     return view_callable(request)
2020-08-18T02:24:10+0000 [stderr#error]   File "/vagrant/weasyl/controllers/settings.py", line 536, in manage_folders_
2020-08-18T02:24:10+0000 [stderr#error]     folder.select_flat(request.userid),
2020-08-18T02:24:10+0000 [stderr#error]   File "/vagrant/weasyl/folder.py", line 188, in select_flat
2020-08-18T02:24:10+0000 [stderr#error]     folders = select_list(userid)
2020-08-18T02:24:10+0000 [stderr#error]   File "/vagrant/weasyl/folder.py", line 179, in select_list
2020-08-18T02:24:10+0000 [stderr#error]     children = folders_by_parentid.get(top_level_folder.folderid)
2020-08-18T02:24:10+0000 [stderr#error] AttributeError: 'dict' object has no attribute 'folderid'
```
And the ``folder`` table is simply (after ``truncate folder cascade`` to verify that this was a thing):
```
weasyl=# select * from folder;
 folderid | parentid | userid | title | settings 
----------+----------+--------+-------+----------
        2 |        0 |      1 | root  | 
(1 row)
```

If I am reading the test suite correctly, I don't think the method we're using to create submissions would trigger this code line, so I think that's why the tests "pass".

Comment (kfkitsune) [reply weasyl/folder.py:179]: Just to note, using the restore from Charmander via Gitter, it still has the above issue.
<details><summary>(Expand for full traceback)
2020-08-19T02:10:09+0000 [stderr#error]   File "/vagrant/weasyl/folder.py", line 179, in select_list
2020-08-19T02:10:09+0000 [stderr#error]     children = folders_by_parentid.get(top_level_folder.folderid)
2020-08-19T02:10:09+0000 [stderr#error] AttributeError: 'dict' object has no attribute 'folderid'</summary>
<pre>
2020-08-19T02:09:52+0000 [-] WeasylSite starting on 8880
2020-08-19T02:09:52+0000 [weasyl.polecat.WeasylSite#info] Starting factory <weasyl.polecat.WeasylSite instance at 0x7fadc84d71b8>
2020-08-19T02:10:09+0000 [stdout#info] unhandled error (request id None) in {'HTTP_ACCEPT': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8', 'webob._parsed_query_vars': (GET([]), ''), 'CONTENT_TYPE': '', 'HTTP_REFERER': 'https://lo.weasyl.com:8443/submit', 'HTTP_X_FORWARDED_PROTO': 'https', 'wsgi.multithread': True, 'SCRIPT_NAME': '', 'HTTP_X_REQUEST_STARTED_AT': '1597803009.70857501', 'REQUEST_METHOD': 'GET', 'HTTP_HOST': 'lo.weasyl.com:8443', 'PATH_INFO': '/submit/visual', 'SERVER_PROTOCOL': 'HTTP/1.0', 'QUERY_STRING': '', 'HTTP_UPGRADE_INSECURE_REQUESTS': '1', 'HTTP_X_REAL_IP': '10.0.2.2', 'CONTENT_LENGTH': '', 'HTTP_USER_AGENT': 'Mozilla/5.0 (X11; Linux x86_64; rv:68.0) Gecko/20100101 Firefox/68.0', 'HTTP_CONNECTION': 'close', 'HTTP_COOKIE': 'WZL=jXhrq4HQqNDOLR00H4Y0GEGmI0SInIFU9poxIj4boLgkBXPV6VehyNaWlTwUkMJ0; __cfduid=df5cd83296b157b956e682775c1454aa21597716017', 'SERVER_NAME': 'lo.weasyl.com', 'REMOTE_ADDR': '127.0.0.1', 'HTTP_X_FORWARDED_FOR': '10.0.2.2', 'wsgi.errors': <twisted.web.wsgi._ErrorStream object at 0x7fadc5acdc50>, 'wsgi.multiprocess': False, 'HTTP_ACCEPT_LANGUAGE': 'en-US,en;q=0.5', 'wsgi.input': <weasyl.middleware.InputWrap object at 0x7fadc6ea68d0>, 'wsgi.version': (1, 0), 'wsgi.url_scheme': 'https', 'wsgi.run_once': False, 'SERVER_PORT': '8880', 'HTTP_ACCEPT_ENCODING': 'gzip, deflate, br', 'webob._parsed_cookies': ({u'WZL': u'jXhrq4HQqNDOLR00H4Y0GEGmI0SInIFU9poxIj4boLgkBXPV6VehyNaWlTwUkMJ0', u'__cfduid': u'df5cd83296b157b956e682775c1454aa21597716017'}, 'WZL=jXhrq4HQqNDOLR00H4Y0GEGmI0SInIFU9poxIj4boLgkBXPV6VehyNaWlTwUkMJ0; __cfduid=df5cd83296b157b956e682775c1454aa21597716017')}
2020-08-19T02:10:09+0000 [stderr#error] Traceback (most recent call last):
2020-08-19T02:10:09+0000 [stderr#error]   File "/vagrant/weasyl-env/local/lib/python2.7/site-packages/pyramid/tweens.py", line 41, in excview_tween
2020-08-19T02:10:09+0000 [stderr#error]     response = handler(request)
2020-08-19T02:10:09+0000 [stderr#error]   File "/vagrant/weasyl/middleware.py", line 161, in query_debug_tween
2020-08-19T02:10:09+0000 [stderr#error]     return handler(request)
2020-08-19T02:10:09+0000 [stderr#error]   File "/vagrant/weasyl/middleware.py", line 233, in http2_server_push
2020-08-19T02:10:09+0000 [stderr#error]     resp = handler(request)
2020-08-19T02:10:09+0000 [stderr#error]   File "/vagrant/weasyl/middleware.py", line 193, in database_session_cleanup_tween
2020-08-19T02:10:09+0000 [stderr#error]     return handler(request)
2020-08-19T02:10:09+0000 [stderr#error]   File "/vagrant/weasyl/middleware.py", line 44, in cache_clear_tween
2020-08-19T02:10:09+0000 [stderr#error]     return handler(request)
2020-08-19T02:10:09+0000 [stderr#error]   File "/vagrant/weasyl/middleware.py", line 63, in db_timer_tween
2020-08-19T02:10:09+0000 [stderr#error]     resp = handler(request)
2020-08-19T02:10:09+0000 [stderr#error]   File "/vagrant/weasyl/middleware.py", line 117, in session_tween
2020-08-19T02:10:09+0000 [stderr#error]     return handler(request)
2020-08-19T02:10:09+0000 [stderr#error]   File "/vagrant/weasyl/middleware.py", line 180, in status_check_tween
2020-08-19T02:10:09+0000 [stderr#error]     return handler(request)
2020-08-19T02:10:09+0000 [stderr#error]   File "/vagrant/weasyl-env/local/lib/python2.7/site-packages/pyramid/router.py", line 148, in handle_request
2020-08-19T02:10:09+0000 [stderr#error]     registry, request, context, context_iface, view_name
2020-08-19T02:10:09+0000 [stderr#error]   File "/vagrant/weasyl-env/local/lib/python2.7/site-packages/pyramid/view.py", line 667, in _call_view
2020-08-19T02:10:09+0000 [stderr#error]     response = view_callable(context, request)
2020-08-19T02:10:09+0000 [stderr#error]   File "/vagrant/weasyl-env/local/lib/python2.7/site-packages/pyramid/config/views.py", line 169, in __call__
2020-08-19T02:10:09+0000 [stderr#error]     return view(context, request)
2020-08-19T02:10:09+0000 [stderr#error]   File "/vagrant/weasyl-env/local/lib/python2.7/site-packages/pyramid/config/views.py", line 188, in attr_view
2020-08-19T02:10:09+0000 [stderr#error]     return view(context, request)
2020-08-19T02:10:09+0000 [stderr#error]   File "/vagrant/weasyl-env/local/lib/python2.7/site-packages/pyramid/config/views.py", line 214, in predicate_wrapper
2020-08-19T02:10:09+0000 [stderr#error]     return view(context, request)
2020-08-19T02:10:09+0000 [stderr#error]   File "/vagrant/weasyl-env/local/lib/python2.7/site-packages/pyramid/viewderivers.py", line 401, in viewresult_to_response
2020-08-19T02:10:09+0000 [stderr#error]     result = view(context, request)
2020-08-19T02:10:09+0000 [stderr#error]   File "/vagrant/weasyl-env/local/lib/python2.7/site-packages/pyramid/viewderivers.py", line 144, in _requestonly_view
2020-08-19T02:10:09+0000 [stderr#error]     response = view(request)
2020-08-19T02:10:09+0000 [stderr#error]   File "/vagrant/weasyl/controllers/decorators.py", line 21, in inner
2020-08-19T02:10:09+0000 [stderr#error]     return view_callable(request)
2020-08-19T02:10:09+0000 [stderr#error]   File "/vagrant/weasyl/controllers/content.py", line 40, in submit_visual_get_
2020-08-19T02:10:09+0000 [stderr#error]     folder.select_flat(request.userid),
2020-08-19T02:10:09+0000 [stderr#error]   File "/vagrant/weasyl/folder.py", line 188, in select_flat
2020-08-19T02:10:09+0000 [stderr#error]     folders = select_list(userid)
2020-08-19T02:10:09+0000 [stderr#error]   File "/vagrant/weasyl/folder.py", line 179, in select_list
2020-08-19T02:10:09+0000 [stderr#error]     children = folders_by_parentid.get(top_level_folder.folderid)
2020-08-19T02:10:09+0000 [stderr#error] AttributeError: 'dict' object has no attribute 'folderid'
</pre>
</details>

Comment (charmander) [reply weasyl/folder.py]: Yep, a broken leftover! Thanks for finding it.

**Comment by charmander** [comment]
Thanks for the review!

**Pull request merged**